tests/commands/test-kob-install.sh

```diff
@@ -1 +1,319 @@
+  
+  __test_kob_install_dummyenv_script > $path_to_kob_envs/kobman-$kobman_env_name.sh
+  
+  touch $path_to_kob_env_tests/test-kobman-${kobman_env_name}.sh
+  __test_kob_dummyenv_validation > $path_to_kob_env_tests/test-kobman-${kobman_env_name}.sh
+  
+  source $path_to_kob_envs/kobman-$kobman_env_name.sh
+
+
+}
+
+function __test_kob_execute {
+  echo "executing install command..."
+  kob install --environment ${kobman_env_name} --version $version > ~/output.txt
+#   echo "output.txt"
+#   cat ~/output.txt
+   
+  cat ~/output.txt | grep -q "dummyenv installed"
+  
+  if [[ "$?" == "0" ]]; then
+  
+    echo "0" > $KOBMAN_DIR/var/kobman_env_$kobman_env_name.proc
+  
+  else
+  
+    echo "1" > $KOBMAN_DIR/var/kobman_env_$kobman_env_name.proc    
+  
+  fi
+}
+
+
+function __test_kob_validate {
+  
+  echo "validating install command...."
+  . $path_to_kob_env_tests/test-kobman-${kobman_env_name}.sh  > ~/tmp.txt
+  
+
+  
+  if [[ $(cat $KOBMAN_DIR/var/kobman_env_$kobman_env_name.proc) == "1" ]]; then
+    
+    echo "install command did not execute properly"
+    status="fail"
+    __test_kob_cleanup
+    __test_kob_install_output
+    return 1
+  
+  fi
+
+  tail -1 ~/tmp.txt | grep -q "success"
+  if [[ "$?" != "0" ]]; then
+  
+    echo "output of dummy env validate is: failed"
+    status="fail"
+    __test_kob_cleanup
+    __test_kob_install_output
+    return 1
+  fi
+  
+  if [[ ! -d ~/Dev_${kobman_env_name} ]]; then
+    
+    echo "Dev_${kobman_env_name} was not found in the /home folder"
+    status="fail"
+    __test_kob_cleanup
+    __test_kob_install_output
+    return 1
+  
+  fi
+  
+  if [[ ! -d $KOBMAN_DIR/envs/kob_env_${kobman_env_name} || $(cat $KOBMAN_DIR/envs/kob_env_${kobman_env_name}/current) != $version ]]; then
+    
+    echo "error in .kobman/envs"
+    status="fail"
+    __test_kob_cleanup
+    __test_kob_install_output
+    return 1
+  
+  fi
+  
+}
+
+
+
+function __test_kob_install_output
+{
+  
+  if [ $status == "true" ]; then
+  
+    echo "test_kob_install success"
+  
+  else  
+  
+    echo "test_kob_install failed"
+  
+  fi
+}
+
+function __test_kob_cleanup()
+{
+  
+  rm ~/output.txt ~/tmp.txt $path_to_kob_env_tests/test-kobman-${kobman_env_name}.sh $path_to_kob_envs/kobman-$kobman_env_name.sh $KOBMAN_DIR/var/kobman_env_$kobman_env_name.proc
+  rm -rf ~/Dev_dummyenv 
+  rm -rf $path_to_kob_envs/kob_env_$kobman_env_name
+
+}
+
+
+
+
+function __test_kob_install_dummyenv_script(){
+  
+cat <<EOF
+#!/bin/bash
+
+function __kobman_install_dummyenv
+{
+  __kobman_install
+  __kobman_uninstall
+  __kobman_update
+  __kobman_upgrade
+  __kobman_start
+  __kobman_stop
+}
+function __kobman_install
+{
+  
+  cd ~
+  if [ ! -d "Dev_dummyenv" ]; then 
+    
+    __kobman_create_dev_environment 
+    __kobman_echo_white "dummyenv installed"
+  else
+    
+    rm -rf ~/Dev_dummyenv	
+    __kobman_create_dev_environment  
+    __kobman_echo_white "dummyenv installed"
+    
+  fi		
+}
+function __kobman_uninstall
```

Comment:
function name should be __kobman_XXXX_YYYY , where XXXX is command and YYYY is environment name.